tests/Controller/TalksControllerTest.php

```diff
@@ -869,18 +895,24 @@ public function testCommentOnTalk(
 
         $talks_comment_email->method('sendEmail');
 
-        $talks_controller = $this->getMockBuilder(TalksController::class)
-            ->setMethods(['getTalkCommentEmailService'])
-            ->getMock();
+        $talks_controller = new class(new NullSpamCheckService(), $talks_comment_email) extends TalksController {
```

Comment:
Using an anonymous class to override the behaviour of the `getTalkCommentEmailService()` method makes a lot more sense here than trying to create a partial mock.

Of course, this would be a lot better if we injected all dependencies instead of creating them in services themselves.

From the general test setup it's not even clear what the system under test is.